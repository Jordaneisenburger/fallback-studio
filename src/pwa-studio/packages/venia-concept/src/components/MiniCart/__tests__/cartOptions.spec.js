import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import CartOptions from '../cartOptions';

const renderer = new ShallowRenderer();

test('it renders the correct tree', () => {
    const props = {
        cartItem: {
            item_id: 99,
            name: 'cartItem name',
            price: 99,
            qty: 99
        },
        configItem: {
            __typename: 'simple'
        },
        currencyCode: 'USD',
        endEditItem: jest.fn(),
        updateCart: jest.fn()
    };

    const tree = renderer.render(<CartOptions {...props} />);

    expect(tree).toMatchSnapshot();
});
