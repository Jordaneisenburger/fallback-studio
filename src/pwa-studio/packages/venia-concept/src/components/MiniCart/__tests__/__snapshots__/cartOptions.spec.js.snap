// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the correct tree 1`] = `
<t>
  <div>
    <span>
      cartItem name
    </span>
    <span>
      <mockConstructor
        currencyCode="USD"
        value={99}
      />
    </span>
  </div>
  <div>
    <section>
      <h2>
        <span>
          Quantity
        </span>
      </h2>
      <Classify(Quantity)
        initialValue={99}
        onValueChange={[Function]}
      />
    </section>
  </div>
  <div>
    <Classify(Button)
      onClick={[MockFunction]}
    >
      <span>
        Cancel
      </span>
    </Classify(Button)>
    <Classify(Button)
      disabled={false}
      onClick={[Function]}
      priority="high"
    >
      <span>
        Update Cart
      </span>
    </Classify(Button)>
  </div>
  <div>
    <LoadingIndicator>
      Updating Cart
    </LoadingIndicator>
  </div>
</t>
`;
